C/include/simplicity/errorCodes.h

```diff
@@ -65,7 +65,7 @@ static inline const char * SIMPLICITY_ERR_MSG(simplicity_err err) {
   case SIMPLICITY_ERR_STOP_CODE:
     return "Program has STOP node";
   case SIMPLICITY_ERR_HIDDEN:
-    return "Program has illegal HIDDEN child node";
+    return "Program has node with a HIDDEN child in a position where it is not allowed";
```

Comment:
4ef6fc264e7235485f5ad3b13eaf0746cb7777cc: I think the message doesn't tell the developer how to fix the error. Non-case nodes may not have hidden children. Case nodes may have up to one hidden child.

Maybe "Hidden node is not part of a well-formed assertion" This is close to @apoelstra 's proposal from #205

Reply:
In a Simplicity DAG, a hidden node can have multiple parents. A hidden node can be part of a well-formed assertion and, at the same time, part of a "not well-formed assertion".

I start to like Russell's proposed error message.